lighthouse/src/main.rs

```diff
@@ -197,6 +197,25 @@ fn run<E: EthSpec>(
         optional_testnet_config = clap_utils::parse_testnet_dir(matches, "testnet-dir")?;
     };
 
+    // WARNING: Temporary migration code for directory restructure
+    // Remove after reasonably sure most users have migrated.
+    let default_root_dir = dirs::home_dir()
+        .map(|home| home.join(DEFAULT_ROOT_DIR))
+        .unwrap_or_else(|| PathBuf::from("."));
+    let testnet_dir = default_root_dir.join(directory::get_testnet_dir(matches));
+
+    if !matches.is_present("datadir") && !testnet_dir.exists() {
+        std::fs::create_dir_all(&testnet_dir)
+            .map_err(|e| format!("Failed to create testnet directory: {}", e))?;
+        for dir in ["validators", "wallets", "secrets", "beacon"].iter() {
```

Comment:
@michaelsproul even if we move the `validator_definitions.yaml`, the keystore paths and keystore password paths within the file will point to wrong directories since they are absolute paths. 
If we delete the definitions file, then the vc will repopulate it but users would loose their manually entered passwords from the `voting_keystore_password` key in the yaml.

So should we be changing the contents of the definitions file and then move the file to reflect the new paths as well as part of the migration?

Reply:
Oh damn, that's a good point. We could attempt an auto-migration but maybe it should be non-destructive (i.e. don't delete the original files -- or move them out of the way).

We should probably also ship the release with a big warning about possibly having to manually intervene to fix things if the automatic migration fails.

Reply:
Summoning @paulhauner and @AgeManning

Reply:
Added a hacky way to migrate the `validator_definitions.yaml` in https://github.com/sigp/lighthouse/pull/1532/commits/ad22178ed6d4591f3fbeef4d42e3438b8904afd2

Reply:
Yeah. I'm of the opinion that software is still experimental. We can do huge breaking releases. 

Try and hack the upgrade but warn people. Best to rip the band-aid off early imo.